code/game/objects/items/weapons/implants/implant_freedom.dm

```diff
@@ -8,14 +8,13 @@
 
 
 /obj/item/weapon/implant/freedom/activate()
-	if(uses == 0)
-		return 0
-	if(uses != -1)
-		uses--
+	uses--
 	imp_in << "You feel a faint click."
 	if(iscarbon(imp_in))
 		var/mob/living/carbon/C_imp_in = imp_in
 		C_imp_in.uncuff()
+	if(!uses)
+		qdel(src)
```

Comment:
When the implant is out of use we delete it which removes the action button from the mob.